src/main/java/com/google/jenkins/plugins/credentials/oauth/JsonServiceAccountConfig.java

```diff
@@ -53,36 +62,93 @@
   private static final long serialVersionUID = 6818111194672325387L;
   private static final Logger LOGGER =
       Logger.getLogger(JsonServiceAccountConfig.class.getSimpleName());
-  private String jsonKeyFile;
+  @Nonnull
+  private final String filename;
+  @Nonnull
+  private final SecretBytes secretJsonKey;
+  @Deprecated   // for migration purpose
+  @CheckForNull
+  private transient String jsonKeyFile;
   private transient JsonKey jsonKey;
 
+  /**
+   * @param jsonKeyFile uploaded json key file
+   * @param filename
+   *     previous json key file name.
+   *     used if jsonKeyFile is not provided.
+   * @param secretJsonKey
+   *     previous json key file content.
+   *     used if jsonKeyFile is not provided.
+   * @since 0.7
+   */
   @DataBoundConstructor
   public JsonServiceAccountConfig(FileItem jsonKeyFile,
-      String prevJsonKeyFile) {
+      String filename, SecretBytes secretJsonKey) {
```

Comment:
If `filename` isn't required, then it should be moved to a setter annotated with `@DataboundSetter`

this would resolve the issue described here:
https://github.com/jenkinsci/google-oauth-plugin/pull/19#issuecomment-475327911

Mandatory arguments should be in the `@DataBoundConstructor` and optional arguments populated via setters

Reply:
Just for completeness: https://jenkins.io/doc/developer/plugin-development/pipeline-integration/#constructor-vs-setters